core/src/com/unciv/logic/automation/SpecificUnitAutomation.kt

```diff
@@ -311,18 +311,19 @@ object SpecificUnitAutomation {
 
     // This really needs to be changed, to have better targetting for missiles
     fun automateMissile(unit: MapUnit) {
-
-        val tilesInRange = unit.currentTile.getTilesInDistance(unit.getRange() * 2)
-
-        for (tile in tilesInRange) {
-            // For now AI will only use nukes against cities because in all honesty that's the best use for them.
-            if (tile.isCityCenter() && tile.getOwner()!!.isAtWarWith(unit.civInfo)) {
-                Battle.nuke(MapUnitCombatant(unit), tile)
-                return
+        val tilesInRange = unit.currentTile.getTilesInDistance(unit.getRange())
+        if (unit.hasUnique("Nuclear weapon")) {
+            for (tile in tilesInRange) {
+                // For now AI will only use nukes against cities because in all honesty that's the best use for them.
+                if (tile.isCityCenter() && tile.getOwner()!!.isAtWarWith(unit.civInfo)) {
+                    Battle.nuke(MapUnitCombatant(unit), tile)
+                    return
+                }
             }
+        } else { //Guided Missile
+            if (BattleHelper.tryAttackNearbyEnemy(unit)) return
```

Comment:
Won't that mean the AI will vastly underestimate the benefit/cost ratio? That `BattleHelper.tryAttackNearbyEnemy` function will still weigh options under the assumption the attacker has a chance to survive - if you haven't changed it, too... Or is my thinking way off?

Reply:
Thx for the suggestions. Here is what tryAttackNearbyEnemy() does as far as I understand :
1. Find attackable enemy units/cities, ignore those whose counter attack will kill the attacking unit
2. Try capture enemy city
3. If 2 fails, attack the most vulnerable enemy unit
4. If 3 fails, attack enemy cities (with health left)

This is absolutely not best strategy but makes sense for AI player to begin with. In my opinion, AIs should avoid using non-nuclear missile in most situations, because benefit/cost ratio is very tricky anyway.

Reply:
For step 1, I think it's OK if AI ignores the fact missile destroys its-self for attacking. Any better advise is welcomed. 😄

Reply:
Just food for thought - at most I would have inserted a `//todo`, but we already have enough of those :)